src/migrations/FolderMigration.ts

```diff
@@ -1,25 +1,37 @@
-import PathResolver from '../r2mm/manager/PathResolver';
 import * as path from 'path';
 import FsProvider from '../providers/generic/file/FsProvider';
-import FileUtils from '../utils/FileUtils';
+import PathResolver from '../r2mm/manager/PathResolver';
 import CacheUtil from '../r2mm/mods/CacheUtil';
+import FileUtils from '../utils/FileUtils';
 
-export default class FolderMigration {
+/**
+ * Mod directory structure was changed when support for other games
+ * besides RoR2 was added. Update the dir structure if the old one is
+ * still in use.
+ */
+export class FolderMigration {
 
-    public static async needsMigration(): Promise<boolean> {
+    public static async needsMigration() {
         const fs = FsProvider.instance;
         return await fs.exists(path.join(PathResolver.ROOT, "mods"));
     }
 
-    public static async runMigration(): Promise<void> {
-        console.log("Started migration");
+    public static async runMigration() {
+        if (!await this.needsMigration()) {
+            return;
+        }
+
+        console.log("Started legacy directory migration");
         const fs = FsProvider.instance;
+        const rorPath = path.join(PathResolver.ROOT, "RiskOfRain2");
         await CacheUtil.clean();
-        if ((await fs.exists(path.join(PathResolver.ROOT, "RiskOfRain2")))) {
-            await FileUtils.emptyDirectory(path.join(PathResolver.ROOT, "RiskOfRain2"));
-            await fs.rmdir(path.join(PathResolver.ROOT, "RiskOfRain2"));
+
+        if (await fs.exists(rorPath)) {
+            await FileUtils.emptyDirectory(rorPath);
+            await fs.rmdir(rorPath);
         }
-        await fs.rename(path.join(PathResolver.ROOT, "mods"), path.join(PathResolver.ROOT, "RiskOfRain2"));
-    }
 
+        await fs.rename(path.join(PathResolver.ROOT, "mods"), rorPath);
+        console.log("Directory migration done");
```

Comment:
What's the policy with console.logging in this project? My initial reaction was to clean them out. Then I thought they might help users to debug stuff. But then again, they don't have access to the console in production, do they?

Reply:
Logging is accessible during production. `Ctrl + Shift + I` for r2modman, and accessible via logs in TMM.

Generally the rule I've mostly followed is that if a user can't see it, we should still be able to track progress via logs.